### PR TITLE
SW-18711 - 5.3 - Async JS

### PR DESCRIPTION
**Pull request by Phil23**

### CODE REVIEW

File: source/designers-guide/css-and-js-files-usage/index.md [hunk 1]
```diff
@@ -1,16 +1,20 @@
 ---
 layout: default
-title: Using CSS and JavaScript files in themes.
+title: Using CSS and JavaScript in themes.
```
Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: please remove the dot (`.`) in the title

File: source/designers-guide/css-and-js-files-usage/index.md [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
-menu_title: Using CSS and JavaScript files
+menu_title: Using CSS and JavaScript
 menu_order: 50
 ---
 
-This quick tip shows off the best way on how to use CSS and JavaScript files for your custom themes, in order to enable them to be automatically compressed by the Shopware theme compiler. To use this feature you have to place your CSS and JavaScript files inside your theme directory under the subdirectories `frontend/_public`. This would be an example directory structure:
+<div class="toc-list"></div>
 
+## The theme compiler
+Shopware uses a compiler to minify and concatenate resource files like CSS and JavaScript to single compressed files. This reduces file size and the amount of server requests. It is highly recommended to also adding your files to the compiler. The generated files are added to the template automatically, so you don't have to worry about including your files at all.
```
Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: I would switch the order of minify and concatenate to clarify the workflow.

Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: The reduces **the** file size and the  amount of server requests.

Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: It is highly recommended to add your files to the compiler, too.

File: source/designers-guide/css-and-js-files-usage/index.md [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
+
+Before you add your files to the theme compiler you have to place them in the correct directory structure `frontend/_public/` inside your custom theme. Here is an example directory structure:
```
Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: I would get rid of the example `frontend/_public/` here as you already show a tree below.

File: source/designers-guide/css-and-js-files-usage/index.md [hunk 4]
```diff
@@ -22,24 +26,44 @@ ExampleTheme
...
+];
 ```
 <div class="alert alert-info" role="alert">
     <strong>Note:</strong> When you add JavaScript with the <code>$javascript</code> array you will also have access to jQuery.
     If you'd like further information about creating your own jQuery plugins you can take a look at our Guide: <a title="Getting started with the statemanager and the jQuery plugin base" href="https://devdocs.shopware.com/designers-guide/javascript-statemanager-and-pluginbase/">Getting started with the statemanager and the jQuery plugin base</a>
 </div>
 
-After clearing the theme cache the changes should be displayed in the storefront.
+### Compiling files
+Every time you add changes to your files the corresponding theme files have to be compiled. There are two ways of doing this.
```
Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: To which files? You should mention css/less/javascript here.

File: source/designers-guide/css-and-js-files-usage/index.md [hunk 5]
```diff
@@ -22,24 +26,44 @@ ExampleTheme
...
 <div class="alert alert-info" role="alert">
     <strong>Note:</strong> When you add JavaScript with the <code>$javascript</code> array you will also have access to jQuery.
     If you'd like further information about creating your own jQuery plugins you can take a look at our Guide: <a title="Getting started with the statemanager and the jQuery plugin base" href="https://devdocs.shopware.com/designers-guide/javascript-statemanager-and-pluginbase/">Getting started with the statemanager and the jQuery plugin base</a>
 </div>
 
-After clearing the theme cache the changes should be displayed in the storefront.
+### Compiling files
+Every time you add changes to your files the corresponding theme files have to be compiled. There are two ways of doing this. 
+
+During production mode you can start the compiler via the administration panel in the cache and performance module. Go to `Configuration` -> `Caches & Performance` -> `Caches` and check the `Compile themes` option.
```
Comment (janbuecker) [inline source/designers-guide/css-and-js-files-usage/index.md]: Please add a headline for each way

**Comment by Phil23** [comment]
@janbuecker all done!

**Pull request merged**